modules/AlphaWalletFoundation/AlphaWalletFoundation/Transactions/TransactionsService.swift

```diff
@@ -142,17 +130,31 @@ public class TransactionsService {
         }
     }
 
-    @objc private func stopTimers() {
+    @objc private func stop() {
         for each in providers {
-            each.value.stopTimers()
+            each.value.stop()
         }
     }
 
-    @objc private func restartTimers() {
+    @objc private func resume() {
         guard !config.development.isAutoFetchingDisabled else { return }
 
         for each in providers {
-            each.value.runScheduledTimers()
+            each.value.resume()
+        }
+    }
+    //TODO: call when receive a push notification
+    public func forceResumeOrStart(server: RPCServer) {
+        guard let provider = providers[server] else { return }
+
+        switch provider.state {
+        case .pending:
+            provider.start()
+        case .running:
+            provider.stop()
```

Comment:
Does it have to stop if it's already running? Is it because it might miss a new transaction?

Reply:
we stop it because in case its running (waiting for next call), that might take some time, (cancelling request shouldn't cause transactions missing)  and running a new cycle (calling resume method) we make force fetch making api calls.
So we reset current cycle in case its running and start new one.